### PR TITLE
Add reducers and actions

### PR DESCRIPTION
**Pull request by manq2010**
Bookstore: add reducers and actions
---

In this PR, the following was done:
- [x] Added reducers and actions for books and categories

### CODE REVIEW

File: src/redux/configureStore.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { applyMiddleware, legacy_createStore as createStore, combineReducers } from 'redux';
+import thunk from 'redux-thunk';
```
Comment (mirouhml) [inline src/redux/configureStore.js]: - [ ] Please know that you need to use `@reduxjs/toolkit` properly here, by importing `configureStore` from it then using it to configure your store, kindly change this.

File: src/redux/books/books.jsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      };
+
+    case EDIT:
+      return {
+
+      };
+    default:
+      return state;
+  }
+};
```
Comment (mirouhml) [inline src/redux/books/books.jsx]: - [ ] Please know that the initial state should be an empty array, kindly fix that.

File: src/redux/categories/categories.jsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        ...state, action.payload,
+      ];
+    case STATUS:
+      return [
+        ...state, action.payload,
+      ];
+    default:
+      return state;
+  }
+};
```
Comment (mirouhml) [inline src/redux/categories/categories.jsx:29]: - [ ] Please know that you should be using `@reduxjs/toolkit` here as well, as I explained before what you used is the old way of implementing a reducer you need to use the new way using the toolkit, kindly check the lesson again and see how to use the new proper way of `@reduxjs/toolkit`.

**Review by mirouhml — CHANGES_REQUESTED** [review]
# REQUIRED CHANGES 🔴

Hi @manq2010, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- Great job setting up the correct linters ✔️
- Highly organized and clean code ✔️ 👨🏻‍💻
- Professional documentation ✔️
- Correct workflow ✔️
- Meaningful commit messages ✔️

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Nemwel-Boniface — APPROVED** [review]
## Hello @manq2010     :wave: ,
I am Nemwel your reviewer for this PR :man_beard:  

Thank you for meeting all the project requirements good job! :partying_face: 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! :partying_face: 🎉 



## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️ 
Met all the milestone requirements ✔️ 

## [Optional] suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**